### PR TITLE
Fix provider crash on malformed network policy 

### LINKED ISSUE #533: Crash on terraform apply in v0.12.3
_This issue was originally opened by @ceralena as hashicorp/terraform#21970. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._
    
<hr>
    
### Terraform Version

```
v0.12.3
```

### Terraform Configuration Files

It's from a few different modules that it'd be awkward to pull together unless absolutely necessary. For context, I'm using the kubernetes and google providers.

### Debug Output

The `debug log` file here:

https://gist.github.com/ceralena/124e267854b05d535b976494c2529ae6

### Crash Output

The `crash.log` file here:

https://gist.github.com/ceralena/124e267854b05d535b976494c2529ae6

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Additional Context

Not that I know of. I'm relying on local gcloud auth for talking to GCP; that's about it.

### References

### PR DESCRIPTION
**Pull request by alexsomesan**
This change avoids the provider crash caused by incomplete peer blocks on network policy resources.

Fixes #533

```
 make testacc TESTARGS="-run '^TestAccKubernetesNetworkPolicy'"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test "./kubernetes" -v -run '^TestAccKubernetesNetworkPolicy' -timeout 120m
=== RUN   TestAccKubernetesNetworkPolicy_basic
--- PASS: TestAccKubernetesNetworkPolicy_basic (2.86s)
=== RUN   TestAccKubernetesNetworkPolicy_withEgressAtCreation
--- PASS: TestAccKubernetesNetworkPolicy_withEgressAtCreation (0.57s)
=== RUN   TestAccKubernetesNetworkPolicy_importBasic
--- PASS: TestAccKubernetesNetworkPolicy_importBasic (0.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-kubernetes/kubernetes	4.972s
```

### CODE REVIEW

File: kubernetes/structure_network_policy.go [hunk 1]
```diff
@@ -142,23 +145,27 @@ func expandNetworkPolicySpec(in []interface{}) (*v1.NetworkPolicySpec, error) {
...
-				if err != nil {
-					return nil, err
-				}
-				ingresses[i].Ports = *policyPorts
+		if ingress == nil {
+			continue
+		}
+		in, ok := ingress.(map[string]interface{})
+		if !ok {
+			continue
```
Comment (pdecat) [inline kubernetes/structure_network_policy.go]: What about returning an error in that case?

Comment (alexsomesan) [reply kubernetes/structure_network_policy.go]: Done

**Comment by alexsomesan** [comment]
```
TF_ACC=1 ./kubernetes.test -test.parallel 1 -test.v -test.run '^TestAccKubernetesNetworkPolicy' -test.v                                                                                               alex@alexs-macbook-4
=== RUN   TestAccKubernetesNetworkPolicy_basic
--- PASS: TestAccKubernetesNetworkPolicy_basic (3.44s)
=== RUN   TestAccKubernetesNetworkPolicy_withEgressAtCreation
--- PASS: TestAccKubernetesNetworkPolicy_withEgressAtCreation (0.60s)
=== RUN   TestAccKubernetesNetworkPolicy_importBasic
--- PASS: TestAccKubernetesNetworkPolicy_importBasic (0.73s)
PASS
------------------------------------------------------------
```

**Comment by cgriggs01** [comment]
LGTM! 👍

**Pull request merged**